### ISSUE TITLE
Add support for Powershell language

### ISSUE THREAD

**Issue by novogeek** [semgrep/semgrep#6659]
**Is your feature request related to a problem? Please describe.**
Powershell is heavily used by developers and sys admins for various types of task automation activities such as managing enterprise infrastructure via configuration as code, automating build and deploy tasks in CI/CD, managing cloud deployments and cloud resource configurations etc. Ability to identify insecure coding patterns in powershell scripts with Semgrep would greatly help in catching critical (security) bugs much earlier and quicker. 

**Describe the solution you'd like**
Add Semgrep support for parsing powershell scripts so that developers/security professionals can write custom rules and leverage them in their Semgrep ecosystem. The grammar for the language can be found here: https://learn.microsoft.com/en-us/powershell/scripting/lang-spec/chapter-15

**Describe alternatives you've considered**
PSScriptAnalyzer (https://github.com/PowerShell/PSScriptAnalyzer) developed by the Powershell team.

**Comment by aryx** [comment]
There's some tree-sitter grammar for Powershell https://github.com/PowerShell/tree-sitter-PowerShell but it has not been maintained and it looks very incomplete.

**Comment by aryx** [comment]
another one: https://github.com/jrsconfitto/tree-sitter-powershell but also looks incomplete

**Comment by pcgeek86** [comment]
Agreed, it would be great to have PowerShell support in Semgrep. It's cross-platform, runs natively on Linux, inside Docker containers, etc. It's important to have good coding practices for PowerShell automation scripts, and semgrep is an excellent tool to handle that. 👍🏻

**Comment by citronneur** [comment]
Hello Team,

At @airbus-cert we published a Powershell grammar for tree-sitter, and we will be happy to support you to integrate it into semgrep : https://github.com/airbus-cert/tree-sitter-powershell

Having powershell in semgrep could be very interesting from a defensive view.

Thanks in advance,

Have a nice day

**Comment by citronneur** [comment]
Hello team, 

have you make progress on this topic?
I can support you but let me know how.

thanks!

**Comment by aryx** [comment]
No progress unfortunately. If we had customers asking for powershell, things might be different, but right now this is very low priority for us.
Regarding how you can help, there are many steps to follow to add a language, as documented here: https://semgrep.dev/docs/contributing/adding-a-language/

**Comment by citronneur** [comment]
Thanks for your feedback, we will try to initiate something!

**Comment by robpomeroy** [comment]
FWIW I work for a small software house and am leading the process of adopting Semgrep. PowerShell support would be excellent. We're primarily C/C++/C#/Fortran, but use many of the scripting languages for supplemental purposes, especially Python and PowerShell.

**Comment by citronneur** [comment]
Hello Semgrep Team,

Do you think it could be interesting for semgrep to add powershell?
Have a nice day,
